tokenserver/assignment/sqlnode/sql.py

```diff
@@ -233,6 +237,22 @@ def __init__(self, sqluri, create_tables=False, pool_size=100,
     def _get_engine(self, service=None):
         return self._engine
 
+    @property
+    def settings(self):
+        # Normalize the various settings by picking out the
+        # `tokenserver.` settings from the pyramid settings
+        # registry, remove the namespace prefix so that they
+        # match the values that should be passed in as *kw
+        # to the __init__()
+        settings = dict(
+           map(lambda (k, v): (
+               k.replace('tokenserver.', ''), v),
+               filter(lambda e: e[0].startswith('tokenserver.'),
+                      (get_current_registry().settings or {}).items()))
+        ) or self._settings
```

Comment:
Should this `or` be the other way around, i.e. `self._settings or dict(...)`?

I'll also note that this is going to iterate all the keys of `get_current_registry().settings` multiple times per request, once for every access to `self.settings`. Probably fine in practice but it seems wasteful.

Reply:
Well, this would favor the thread local registry settings over the kw values passed in during the Node init. If I remember, you said that the registry settings could change at any point, even after init. I agree that it's a bit wasteful. If these aren't as dynamic as I thought, I'd prefer to do this sort of stuff once and then cache the result.

Reply:
> you said that the registry settings could change at any point

I do recall saying this, and it's true, but as mentioned in slack I think I've convinced myself that our use of `get_current_registry()` is entirely unnecessary in practice and could be safely removed.

(There's a whole bunch of indirection that happens via component registration in pyramid, which results in the app always loading the `INodeAssignment` backend from the current registry, and the backend from the current registry should always have been created using the settings from that registry, so there's no reason for us to check them on every request).